src/main/java/ch/njol/skript/lang/function/Functions.java

```diff
@@ -331,7 +300,8 @@ public static void unregisterFunction(Signature<?> signature) {
 		while (namespaceIterator.hasNext()) {
 			Namespace namespace = namespaceIterator.next();
 			if (namespace.removeSignature(signature)) {
-				globalFunctions.remove(signature.getName());
+				if (!signature.local)
+					globalFunctions.remove(signature.getName());
 
```

Comment:
local functions don't their own map, they're retrieved using the namespaces map

Reply:
It looks like it's all screwed up. ORIGIN.JAVA isn't even being used as a key. globalFunctions and javaNamespace should be removed, and referred to by a Namespace.KEY and in this pull request you've overridden the ORIGIN.SCRIPT with your local functions. The SCRIPT one could be named to GLOBAL and you should be using a key called like LOCAL or something, and the Java made functions should be stored in that namespace rather than it's own field.

Reply:
> ORIGIN.JAVA isn't even being used as a key

It is being used
![image](https://user-images.githubusercontent.com/98935832/194235534-3a42f581-2662-4bdb-b3de-343d6363f0f5.png)

> globalFunctions and javaNamespace fields should be removed, and referred to by their proper Namespace.KEY

They have their own map to make lookup instant, so I don't believe they should be removed

> in this pull request you've overridden the ORIGIN.SCRIPT with your local functions which was supposed to be for general functions.

The Origin enum just tells you whether the function was registered in java (built-in) or in a script (custom, made by players), so it is used correctly by local functions

> and the Java made functions should be stored in the JAVA namespace rather than it's own field.

They are stored in the namespace with the JAVA origin, there's just a field that references it
#
Anyway, I don't think this is the right PR to do so, maybe a function rework PR would be a better way to do it